src/elm/Data/DataSource.elm

```diff
@@ -0,0 +1,102 @@
+module Data.DataSource
+    exposing
+        ( DataSourceId(..)
+        , DataSource
+        , decode
+        , decodeList
+        , encodeId
+        )
+
+import Json.Decode as Decode exposing (Decoder)
+import Json.Encode
+
+
+type DataSourceId
```

Comment:
Thoughts on naming this type `Id` and having other modules refer to it with the namespace like `DataSource.Id` ?